tremor-script/src/grammar.lalrpop

```diff
@@ -1250,7 +1250,34 @@ Merge: MergeRaw<'input> = {
 // Merges a perge spec (record) into the target
 
 For: ComprehensionRaw<'input, ExprRaw<'input>> = {
-    <start:@L> "for" <target:ComplexExprImut> "of" <cases:ForCaseClauses> "end" <end:@L> => ComprehensionRaw{target, cases, mid: NodeMeta::new_box(start, end)},
+    <start:@L> "for" <target:ComplexExprImut> "of" <cases:ForCaseClauses> "end" <end:@L> => ComprehensionRaw{
+        target,
+        cases,
+        mid: NodeMeta::new_box(start, end),
+        initial: ImutExprRaw::Literal(LiteralRaw{value: Value::array(), mid: NodeMeta::new_box(start, end)}),
+        fold: ComprehensionFoldOpRaw(BinOpKind::Add)
+    },
+    <start:@L> "for" <target:ComplexExprImut> "of" <cases:ForCaseClauses> "into" <initial:ComplexExprImut> "end" <end:@L> => ComprehensionRaw{
+        target,
+        cases,
+        mid: NodeMeta::new_box(start, end),
+        initial,
+        fold: ComprehensionFoldOpRaw(BinOpKind::Add)
+    },
+    <start:@L> "for" <target:ComplexExprImut> "of" <cases:ForCaseClauses> "into" <initial:ComplexExprImut> "use" <fold:FoldOperator> "end" <end:@L> => ComprehensionRaw{
```

Comment:
can't this be solved with 1 rule here where the `use` and `into` clauses are optional? That is, are marked with `?`

Reply:
I've reduced it to too, as it simplifies the code in the implementation